File: analogagenda.client/src/app/DTOs/note-entry.dto.ts
import { NoteEntryRuleDto } from './note-entry-rule.dto';
import { NoteEntryOverrideDto } from './note-entry-override.dto';

export interface NoteEntryDto {
  id: string;
  noteId: string;
  time: number;
  step: string;
  details: string;
  index: number;
  temperatureMin: number;
  temperatureMax?: number;
  rules: NoteEntryRuleDto[];
  overrides: NoteEntryOverrideDto[];
}

File: Database/DTOs/NoteEntryDto.cs
namespace Database.DTOs;

public class NoteEntryDto
{
    public string Id { get; set; } = string.Empty;
    public required string NoteId { get; set; }
    public double Time { get; set; }
    public required string Step { get; set; }
    public string Details { get; set; } = string.Empty;
    public int Index { get; set; }
    public double TemperatureMin { get; set; }
    public double? TemperatureMax { get; set; }
    public List<NoteEntryRuleDto> Rules { get; set; } = [];
    public List<NoteEntryOverrideDto> Overrides { get; set; } = [];
}
